.github/workflows/github-ci.yml

```diff
@@ -0,0 +1,48 @@
+name: GitHub CI
+
+on:
+  push:
+    branches: [ master, nightly ]
```

Comment:
I think we need to add the coverity stuff to this as well. Not sure how hard it is to integrate manually...

Reply:
I can try adding coverity after this is merged. It needs to be done directly in master, so that it's triggered by a push to nightly.